### DISCUSSION TITLE
GOS by MTEX

### DISCUSSION THREAD

**Discussion by alielasheri** [mtex-toolbox/mtex#1071] [Ask Anything] [answered]
Hello everyone,
![GOS](https://user-images.githubusercontent.com/74800522/120488454-ca2d5500-c384-11eb-8fb2-3a31dd1c683b.jpg)
I am currently working on the microstructure that evolved during hot deformation to have information about dynamic recrystallization using GOS maps. Actually, I am very beginner at using MTEX so I am seeking your help to answer two questions I have;

the first one is about the grains calculation line:
```matlab
[grains, ebsd.grainId] = calcGrains(ebsd('indexed'),'threshold',7.5*degree);
```
what are the criteria to select the threshold degree here? providing that I am looking for the recrystallized grains (GOS<2 degrees).

the second question:

Can I add the other angle boundaries to the plot ( e.g., low angle boundaries)

I attached a figure showing the GOS map I got by MTEX. I coloured the grains with GOS<2 degrees in white

I wish I could explain my issue in a proper way.

Ali

**Comment by ralfHielscher** [comment]
For the second question have a look at https://mtex-toolbox.github.io/SubGrainBoundaries.html

For the first question I would simply try different threshold and check what gives the best result.

Ralf.

**Reply by alielasheri** [reply]
Thank you So much for your answer,

for the second question, could you please help me how can I add these boundaries on GOS map not on the IPF map.

here is the code I use currently:

% path to files
pname = 'C:\others\mtex-5.6.1\mtex-5.6.1';

% which files to be imported
fname = [pname '\0.4Si_450_L1-2.crc'];

%% Import the Data

% create an EBSD variable containing the data
ebsd = EBSD.load(fname,CS,'interface','crc',...
  'convertEuler2SpatialReferenceFrame');

% consider only indexed data
ebsd = ebsd('indexed');

[grains, ebsd.grainId] = calcGrains(ebsd('indexed'));
ebsd(grains(grains.grainSize < 5)) = [];
[grains, ebsd.grainId] = calcGrains(ebsd('indexed'),'threshold',7.5*degree);
ebsd = ebsd.project2FundamentalRegion;
grains = smooth(grains,5);

% plot the data
plot(ebsd, ebsd.orientations)

ori = ebsd(ebsd.grainId == 4).orientations

plot(grains, grains.meanOrientation)


mis2mean = inv(grains(4).meanOrientation) .* ori


mis2mean = calcGROD(ebsd, grains)

% take the avarage of the misorientation angles for each grain
GOS = ebsd.grainMean(mis2mean.angle);

% plot it
plot(grains, GOS ./ degree)
mtexColorbar('title','GOS in degree')
hold on
plot(grains.boundary,'lineWidth',2)
hold off

Thanks again

**Reply by ralfHielscher** [reply]
Hi,

there is no difference in plotting inner boundaries on top of a ipf map or on top of a GOS map.
The command are exactly the same. You can take them directly from the docu page a linked above.

Ralf.

**Reply by alielasheri** [reply]
Hi Dr. Ralf,

I have already made that but the result was the same. before adding the command of GOS, everything is going very well and I got the inner boundaries. but once, I add the GOS command, the inner boundaries disappear.

I will explain with an example 

here is the modified command I used

create an EBSD variable containing the data
ebsd = EBSD.load(fname,CS,'interface','crc',...
  'convertEuler2SpatialReferenceFrame');

% consider only indexed data
ebsd = ebsd('indexed');

[grains, ebsd.grainId] = calcGrains(ebsd('indexed'));
ebsd(grains(grains.grainSize < 5)) = [];

[grains,ebsd.grainId] = calcGrains(ebsd('indexed'),'threshold',[1*degree, 10*degree]);

% lets smooth the grain boundaries a bit
grains = smooth(grains,5)

% plot the ebsd data
plot(ebsd('indexed'),ebsd('indexed').orientations,'faceAlpha',0.5,'figSize','large')

% init override mode
hold on

% plot grain boundares
plot(grains.boundary,'linewidth',2)

% compute transparency from misorientation angle
alpha = grains.innerBoundary.misorientation.angle / (5*degree);

% plot the subgrain boundaries
plot(grains.innerBoundary,'linewidth',1.5,'edgeAlpha',alpha,'linecolor','b');

% stop override mode
hold off

**_until here, everything is fine and I got the inner boundaries, but once I add the following command for GOS, it disappear again,_**

% take the avarage of the misorientation angles for each grain
GOS = ebsd.grainMean(mis2mean.angle);

% plot it
plot(grains, GOS ./ degree)
mtexColorbar('title','GOS in degree')
hold on
plot(grains.boundary,'lineWidth',2)
hold off

**Comment by zmichels** [comment, marked as answer]
Hi there,

Two thoughts…

1. If you try to plot grains colored by something (like GOS, for example), it may plot *over* anything you’ve already started to plot (with the interior of grains filled based on the value you use). I would maybe start the plotting with the grains as the foundation of your plot layers.

2. In the code you posted, you included the following lines of code which will *stop* plotting new things on top of the existing plot, so you may want to remove that line…
> % stop override mode
> hold off
> 


If I were trying to make a plot like the one I *think* you want to make, I’d probably do something like the following:

% open a new figure window
figure,
% Plot the grains, colored by G.O.S., with boundary linewidth of 2
plot(grains, GOS ./ degree, 'lineWidth',2)
% add a colorbar with title
mtexColorbar('title','GOS in degree')
% hold the plot so can add to it
hold on
% add/plot the subgrain boundaries
plot(grains.innerBoundary,'linewidth',1.5,'edgeAlpha',alpha,'linecolor','b');


Hope that helps some.

Cheers,
Zach

**Reply by alielasheri** [reply]
Hi zmichels,

Wow, it works.

All I can say is thank you so much. I really appreciate it

**Reply by pxyg** [reply]
Hello alielasheri,

May I ask how you colored the GOS<2 area into white? I would like to color the GOS area with a cettain color. Thanks!
